src/Hint/Bracket.hs

```diff
@@ -102,9 +108,9 @@ bracket bad = f Nothing
 
         -- f (Maybe (index, parent, gen)) child
         f :: (Data (a S), Uniplate (a S), ExactP a, Pretty (a S), Brackets (a S)) => Maybe (Int,a S,a S -> a S) -> a S -> [Idea]
-        f Just{} o@(remParen -> Just x) | isAtom x = bracketError msg o x : g x
-        f Nothing o@(remParen -> Just x) | bad || isAtom x = (if isAtom x then bracketError else bracketWarning) msg o x : g x
-        f (Just (i,o,gen)) v@(remParen -> Just x) | not $ needBracket i o x =
+        f Just{} o@(remParens -> Just x) | isAtom x = bracketError msg o x : g x
+        f Nothing o@(remParens -> Just x) | bad || isAtom x = (if isAtom x then bracketError else bracketWarning) msg o x : g x
```

Comment:
This isAtom check caused the test on line 12 to flip from Warning to Suggestion. (Removing both sets of parens gives isAtom == false. Before we only removed one so isAtom was true.)

This seemed harmless, and I didn't really understand this logic, so I just changed the test.